internal/policy/policy.go

```diff
@@ -410,6 +429,13 @@ func validateIdentity(identity cosign.Identity) error {
 	return errs
 }
 
+func emptyIdentity(identity cosign.Identity) bool {
```

Comment:
If you're looking to check if all fields in this `identity` struct are their zero value (empty string in the case of these fields), you might consider using the `.isZero()` method from the `reflect` package. This checks to see if the I think this could eliminate the need for this function, as well as change the call on line 191 to something like:

```
if !reflect.ValueOf(opts.Identity).isZero(){
    p.identity = opts.Identity
```

Reply:
Thank you for this!

I started reading up on the reflect package and this caught my eye:
> It panics if the argument is invalid. 

Given that reflection is quite slow and the possibility of panicking, I decided to go with a different approach:

```go
opts.Identity != (cosign.Identity{})
```

That seems safer to me. Let me know what you think.

Reply:
I think this should work in this context as well. :+1: